src/commcare_cloud/commands/terraform/aws.py

```diff
@@ -150,3 +159,116 @@ def run(self, args, unknown_args):
+        default_username)) or default_username
+    mfa_token = input("Enter your MFA token: ")
+    generate_session_profile(aws_profile, username, mfa_token, duration_minutes)
+
+    puts(colored.green(u"✓ Sign in accepted"))
+    puts(colored.cyan(
+        "You will be able to use AWS from the command line for the next {} minutes."
+        .format(duration_minutes)))
+    puts(colored.cyan(
+        "To use this session outside of commcare-cloud, "
+        "prefix your command with AWS_PROFILE={}:session".format(aws_profile)))
+    return aws_session_profile
+
+
+def generate_session_profile(aws_profile, username, mfa_token, duration_minutes):
+    # General idea from https://www.vividcortex.com/blog/setting-up-multi-factor-authentication-with-the-aws-cli
+    boto_session = boto3.session.Session(profile_name=aws_profile)
+    iam = boto_session.client('iam')
+    devices = iam.list_mfa_devices(UserName=username)['MFADevices']
+    device_arn = sorted(devices, key=lambda device: device['EnableDate'])[-1]['SerialNumber']
+    sts = boto_session.client('sts')
+    credentials = sts.get_session_token(
+        SerialNumber=device_arn,
+        TokenCode=mfa_token,
+        DurationSeconds=duration_minutes * 60,
+    )['Credentials']
+    _write_credentials_to_aws_credentials(
+        '{}:session'.format(aws_profile),
+        aws_access_key_id=credentials['AccessKeyId'],
+        aws_secret_access_key=credentials['SecretAccessKey'],
+        aws_session_token=credentials['SessionToken'],
+        expiration=credentials['Expiration'],
+    )
+
+
+def _write_credentials_to_aws_credentials(
+        aws_profile, aws_access_key_id, aws_secret_access_key, aws_session_token,
+        expiration,
+        aws_credentials_path=os.path.expanduser('~/.aws/credentials')):
+    # followed code examples from https://gist.github.com/incognick/c121038dbd2180c683fda6ae5e30cba3
+    config = configparser.ConfigParser()
+    config.read(os.path.realpath(aws_credentials_path))
+    if aws_profile not in config.sections():
+        config.add_section(aws_profile)
+    config.set(aws_profile, 'aws_access_key_id', aws_access_key_id)
+    config.set(aws_profile, 'aws_secret_access_key', aws_secret_access_key)
+    config.set(aws_profile, 'aws_session_token', aws_session_token)
+    config.set(aws_profile, 'expiration', expiration.strftime("%Y-%m-%dT%H:%M:%SZ"))
+    with open(aws_credentials_path, 'w') as f:
+        config.write(f)
+
+
+def _has_valid_session_credentials(
+        aws_profile, aws_credentials_path=os.path.expanduser('~/.aws/credentials')):
+    config = configparser.ConfigParser()
+    config.read(os.path.realpath(aws_credentials_path))
+    if aws_profile not in config.sections():
+        return False
+    try:
+        expiration = datetime.strptime(config.get(aws_profile, 'expiration'), "%Y-%m-%dT%H:%M:%SZ")
+    except configparser.NoOptionError:
+        return False
+
+    return datetime.utcnow() < expiration
```

Comment:
What is the failure mode if this returns true but the credentials are actually no longer valid?